code/game/machinery/Sleeper.dm

```diff
@@ -543,14 +559,243 @@
 		usr.loc = src
 		usr.reset_view()
 		src.occupant = usr
-		update_icon()
-
+		connected.process()
 		for(var/obj/O in src)
 			qdel(O)
 		src.add_fingerprint(usr)
 		if(!(stat & (BROKEN|NOPOWER)))
 			set_light(light_range_on, light_power_on)
+		update_icon()
 		return
 	return
 
-#undef SLEEPER_SOPORIFIC_DELAY
+/obj/machinery/sleeper/mancrowave
+	name = "thermal homeostasis regulator"
+	desc = "This invention by Mancrowave Inc. is meant for stabilising body temperature. Modern medical technology is amazing."
+	icon_state = "mancrowave_open"
+	base_icon = "mancrowave"
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/sleeper/mancrowave,
+		/obj/item/weapon/stock_parts/scanning_module,
+		/obj/item/weapon/stock_parts/manipulator,
+		/obj/item/weapon/stock_parts/manipulator
+	)
+	connected_type = "/obj/machinery/sleep_console/mancrowave_console"
+	setting = "Thermoregulate"
+	available_options = list("Thermoregulate" = 50)
+	light_color = LIGHT_COLOR_ORANGE
+	automatic = 1
+	drag_delay = 0
+
+/obj/machinery/sleeper/mancrowave/go_out(var/exit = src.loc)
+	if(on && !emagged)
+		return 0
+	else
+		on = 0
+		..()
+
+/obj/machinery/sleeper/mancrowave/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	if(istype(W,/obj/item/weapon/card/emag))
```

Comment:
Yes using the machine flag EMAGGABLE similar to WRENCHANCHOR or whatever is the correct behavior, it wasn't replaced everywhere when it was initially added